### PR TITLE
Keep viewports in sync outside of graphical editing scope

### PR DESCRIPTION
**Pull request by pmconne**
BriefcaseConnection buffers changes to geometry guids until:
- saveChanges, undo, redo, or pull changes, outside the context of a GraphicalEditingScope; or
- The GraphicalEditingScope exits

It then synchronizes:
- Invalidate scenes of viewports viewing modified models;
- Purge tile trees for modified models (only necessary outside of GraphicalEditingScope);
- Remove deleted models from IModelConnection.Models.loaded; and
- Update `geometryGuid` property of loaded models.

Merged BriefcasePushPullNotifications into TxnNotifications.
Simplified ConcurrencyControl.
Added BriefcaseTxns events for pushing and pulling changes.
Cleaned up PushPullStatusField to use the new events.

Incidentally:
- Add Id64.iterable; deprecate iterate and forEach
- IModelConnection.Models.loaded should never have been a public mutable map. Deprecate it.

### CODE REVIEW

**Comment by pmconne** [comment]
@swwilson-bsi please review ConcurrencyControl changes

**Comment by pmconne** [comment]
@bsteinbk please review PushPullStatusField changes.

**Review by bsteinbk — APPROVED** [review]

File: core/frontend/src/BriefcaseConnection.ts [hunk 1]
```diff
@@ -15,14 +15,144 @@ import { IModelConnection } from "./IModelConnection";
...
+    const modelIds = new Set<string>(this._modelIdToGuid.keys());
+    for (const deleted of this._deletedModels) {
+      modelIds.add(deleted);
+      models.unload(deleted);
+    }
+
+    this.invalidateScenes(modelIds);
+    disposeTileTreesForGeometricModels(modelIds, this._briefcase);
+
+    this._briefcase.onBufferedModelChanges.raiseEvent(modelIds);
```
Comment (kabentley) [inline core/frontend/src/BriefcaseConnection.ts:127]: should you send this event out after you've cleared the lists?

Comment (pmconne) [reply core/frontend/src/BriefcaseConnection.ts:127]: Unclear to me what difference that would make - can you elaborate?

Comment (kabentley) [reply core/frontend/src/BriefcaseConnection.ts:127]: what happens if something happens in the event that adds to those lists?

**Review by kabentley — APPROVED** [review]

File: core/frontend/src/BriefcaseConnection.ts (hunk 1 above)
Comment (pmconne) [reply core/frontend/src/BriefcaseConnection.ts:127]: The event is used strictly for tests, which do nothing but verify the list of model Ids is as expected.
```
  /** Strictly for tests - dispatched from ModelChangeMonitor.processBuffered.
   * @internal
   */
  public readonly onBufferedModelChanges = new BeEvent<(changedModelIds: Set<string>) => void>();
```

**Review by grigasp — APPROVED** [review]
Approving presentation changes

**Review by markschlosseratbentley — APPROVED** [review]

**Pull request merged**